packages/@aws-cdk/aws-kms/test/key.test.ts

```diff
@@ -966,16 +967,40 @@ describe('key specs and key usages', () => {
     });
   });
 
-  test('invalid combinations of key specs and key usages', () => {
+  test.each([
```

Comment:
This really concerns me. I know you didn't write this original contract but this many invalid options that won't be discovered until synth time makes me uneasy. Is there a contract update we can make here to enforce the valid combos better?

Reply:
Maybe it would be better to create a separate class for Hmac keys? Maybe there actually should be a few classes:

SymmetricKey 
- only one to have enableKeyRotation in props
- usage is always ENCRYPT_DECRYPT
HmacKey
- usage is always GENERATE_VERIFY_MAC
ShaKey
- usage is always SIGN_VERIFY
RsaKey
- usage options SIGN_VERIFY or ENCRYPT_DESCRYPT

But also the base needs some (breaking?) refactorings because I think a key that has no ENCRYPT_DECRYPT usage should not have grantDescrypt/grantEncrypt methods.

Reply:
@Jacco How can we ensure backwards compatibility with this separation?
When someone upgrades to a potential version with a class for each key, how will the Basic Key class behave?

Reply:
@TheRealAmazonKendra Do you mean extending the tests to make the invalid combinations clearer? Or changing the logic inside the production code?